### PR TITLE
Policy Pack for Azure CIS v2.0 Enforcement for Section 8 - Key Vault

### PR DESCRIPTION
**Pull request by bigdatasourav**
- [x] 8.1 to 8.7
- [ ] 8.8 (Pending for custom control)

image

image

image

image

### CODE REVIEW

File: cis_enforcements/azure/cis_v200/section8/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+By default, the policies are set to `Check` in the pack's policy settings. To enable automated enforcements, you can switch these policies settings by adding a comment to the `Check` setting and removing the comment from one of the listed enforcement options:
+
+```hcl
+resource "turbot_policy_setting" "azure_keyvault_vault_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-keyvault#/policy/types/vaultApproved"
+  note     = "Azure CIS v2.0.0 - Control: 8.6 and 8.7"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/README.md:86]:
```suggestion
  # value    = "Check: Approved"
  value    = "Enforce: Delete unapproved if new"
```

File: cis_enforcements/azure/cis_v200/section8/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+}
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/README.md:87]:
```suggestion
}

resource "turbot_policy_setting" "azure_keyvault_key_expiration" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-keyvault#/policy/types/keyExpiration"
  note         = "Azure CIS v2.0.0 - Control: 8.1 and 8.2"
  # value       = "Check: Expiration"
  value        = "Enforce: Expiration"
}

# Azure > KeyVault > Secret > Expiration
resource "turbot_policy_setting" "azure_keyvault_secret_expiration" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-keyvault#/policy/types/secretExpiration"
  note         = "Azure CIS v2.0.0 - Control: 8.3 and 8.4"
  # value        = "Check: Expiration"
  value        = "Enforce: Expiration"
}

# Azure > KeyVault > Vault > PurgeProtection
resource "turbot_policy_setting" "azure_keyvault_vault_purge_protection" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-keyvault#/policy/types/vaultPurgeProtection"
  note         = "Azure CIS v2.0.0 - Control: 8.5"
  # value        = "Check: Purge Protection Enabled"
  value        = "Enforce: Enable Purge Protection"
}
```

File: cis_enforcements/azure/cis_v200/section8/policies.tf [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Azure > KeyVault > Vault > Approved
+resource "turbot_policy_setting" "azure_keyvault_vault_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/azure-keyvault#/policy/types/vaultApproved"
+  note     = "Azure CIS v2.0.0 - Control: 8.6 and 8.7"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/policies.tf:7]:
```suggestion
  # value    = "Enforce: Delete unapproved if new"
```

File: cis_enforcements/azure/cis_v200/section8/policies.tf [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+          "message": "Role based access control is disabled"
+      } -%}
+
+    {%- else -%}
+
+      {%- set data = {
+          "title": "Role Based Access Control",
+          "result": "Approved",
+          "message": "Role based access control is enabled"
+        } -%}
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/policies.tf:41]:
```suggestion
    {%- elif $.vault.enableRbacAuthorization == false -%}

      {%- set data = {
          "title": "Role Based Access Control",
          "result": "Approved",
          "message": "Role based access control is enabled"
        } -%}
        
    {%- else -%}

      {%- set data = {
          "title": "Role Based Access Control",
          "result": "Skip",
          "message": "No data for role based access control yet"
        } -%}
```

File: cis_enforcements/azure/cis_v200/section8/policies.tf [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    {%- endif -%}
+
+    {% set results = results.concat(data) -%}
+
+    {%- if $.vault.privateEndpointConnections == null -%}
+
+      {%- set data = {
+          "title": "Private endpoint connections",
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/policies.tf:50]:
```suggestion
          "title": "Private Endpoint Connections",
```

File: cis_enforcements/azure/cis_v200/section8/policies.tf [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+          "result": "Not approved",
+          "message": "Private endpoint connections is not used"
+      } -%}
+
+    {%- else -%}
+
+      {%- set data = {
+          "title": "Private endpoint connections",
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/policies.tf:58]:
```suggestion
          "title": "Private Endpoint Connections",
```

File: cis_enforcements/azure/cis_v200/section8/policies.tf [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    {%- endif -%}
+
+    {% set results = results.concat(data) -%}
+
+    {%- if $.vault.privateEndpointConnections == null -%}
+
+      {%- set data = {
+          "title": "Private endpoint connections",
+          "result": "Not approved",
+          "message": "Private endpoint connections is not used"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/policies.tf:52]:
```suggestion
          "message": "Private endpoint connections are not used"
```

File: cis_enforcements/azure/cis_v200/section8/policies.tf [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      } -%}
+
+    {%- else -%}
+
+      {%- set data = {
+          "title": "Private endpoint connections",
+          "result": "Approved",
+          "message": "Private endpoint connections is used"
```
Comment (SumitPopat) [inline cis_enforcements/azure/cis_v200/section8/policies.tf:60]:
```suggestion
          "message": "Private endpoint connections are used"
```

File: cis_enforcements/azure/cis_v200/section8/policies.tf (hunk 4 above)
Comment (bigdatasourav) [reply cis_enforcements/azure/cis_v200/section8/policies.tf:41]: The value of `enableRbacAuthorization` is either true or false, so `No data` condition will not occur.

**Review by SumitPopat — APPROVED** [review]

**Pull request merged**